apps/prairielearn/src/ee/pages/instructorInstanceAdminLti13/instructorInstanceAdminLti13.ts

```diff
@@ -74,42 +76,75 @@ router.get(
       throw error.make(404, 'LTI 1.3 instance not found.');
     }
 
+    const assessments = await selectAssessments({
+      course_instance_id: res.locals.course_instance.id,
+    });
+
+    const lti13_assessments = await queryRows(
+      sql.select_lti13_assessments,
+      {
+        lti13_course_instance_id: instance.lti13_course_instance.id,
+      },
+      Lti13AssessmentsSchema,
+    );
+
+    const lti13AssessmentsByAssessmentId: Record<string, Lti13Assessments> = {};
+    for (const a of lti13_assessments) {
+      lti13AssessmentsByAssessmentId[a.assessment_id] = a;
+    }
+
     if ('lineitems' in req.query) {
+      const assessmentsById: Record<string, AssessmentRow> = {};
+      for (const a of assessments) {
+        assessmentsById[a.id] = a;
+      }
+
+      let lineitems: Lineitems;
+
       try {
-        res.end(LineitemsInputs(await getLineitems(instance)));
+        lineitems = await getLineitems(instance);
       } catch (error) {
         res.end(html`<div class="alert alert-warning">${error.message}</div>`.toString());
         logger.error('LineitemsInputs error', error);
+        return;
       }
-      return;
-    }
 
-    const assessments = await queryRows(
-      sql.select_assessments,
-      {
-        course_instance_id: res.locals.course_instance.id,
-        authz_data: res.locals.authz_data,
-        req_date: res.locals.req_date,
-        assessments_group_by: res.locals.course_instance.assessments_group_by,
-      },
-      AssessmentRowSchema,
-    );
+      const lti13AssessmentsByLineItemIdUrl: Record<string, Lti13Assessments> = {};
+      for (const a of lti13_assessments) {
+        lti13AssessmentsByLineItemIdUrl[a.lineitem_id_url] = a;
+      }
```

Comment:
Rather than passing around these extra map data structures, can we instead embed the relevant information in each `lineitems` row?